blog_posts/2024-09-12-optimisations-text-more.markdown

```diff
@@ -0,0 +1,142 @@
+
+### Browser Extension
+There's been a lot of improvements to the extension codebase, and some bug fixes here and there. The major points here though are:
+
+- @danielhjacobs has upgraded us fully to Manifest V3 on Firefox, with a new Onboarding screen for permission requests.
+- @danielhjacobs has also brought back "SWF takeover" (loading a SWF url directly, and have it play in Ruffle instead of downloading) using the upcoming header detection feature of `declarativeNetRequest` in Manifest V3. This API only works in latest Chromium currently.
+- @riku-42 has added support for 4399.com (a Chinese flash game site) with our extension by adding better Flash detection workarounds.
+
+### Web Improvements
+There has also been a lot of improvements under the hood here for code quality, and especially preparation for releasing a stable version. There's a few notable highlights though:
+
+- Thanks to contributions from @WumboSpasm, the general look and feel of the UI on web is much improved. This includes support for dark mode, which will match the browsers settings.
+- @dinnerbone has been working on stabilizing our JavaScript API, and also making a `ruffle-core` package that anyone can use as an npm dependency, instead of relying on our prebuilt zip packages. This work is ongoing but should be completed soon!
+- @dinnerbone has improved `ExternalInterface` support, allowing support for games like Club Penguin which interacted with the page heavily.
+
+### Rendering Improvements
+@Aaron1011 has fixed a bunch of bugs relating to Stage3D, and improved PixelBender support.
+
+@moulins has found and fixed lots of smaller issues with our Filter implementations, and provided better tests for them.
+
+@kjarosh, @Aaron1011 and @adrian17 have improved the ActionScript `Graphics` API, making content that used it look more correct.
+
+
+### Other improvements
+There have been *so many* changes from so many people, it's hard to list them all. Here's a few notable changes that are worth mentioning but are too hard to categorise:
+- @sleepycatcoding, @evilpie, @dinnerbone, @adrian17 and @Aaron1011 have all improved the AVM2 XML API to some degree - we can now handle most advanced uses, which significantly improves support of games that use it.
+- @kjarosh has implemented support for right-clicking and middle-clicking in content, implemented mouse wheel events, and improved double-click and triple-click support (especially with text selection)
+- @Lord-McSweeney has implemented "Loader reuse", which unblocked many online/MMO games which relied on this.
+- @dinnerbone has implemented `LocalConnection` support, which makes Club Penguin work in Ruffle.
+- @evilpie has implemented File Selection APIs used by Scratch, allowing you to download and upload files in content that uses it.
+... and so so so many bug fixes and miscellaneous improvements!

```

Comment:
I believe this was meant to be in the next paragraph, it needs an additional newline before it.